src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -370,38 +368,33 @@ async def coursier_resolve_lockfile(
     )
 
     coursier_report_file_name = "coursier_report.json"
+
     process_result = await Get(
         ProcessResult,
-        Process(
-            argv=coursier.args(
-                [
-                    coursier_report_file_name,
-                    *coursier_resolve_info.coord_arg_strings,
-                    # TODO(#13496): Disable --strict-include to work around Coursier issue
-                    # https://github.com/coursier/coursier/issues/1364 which erroneously rejects underscores in
-                    # artifact rules as malformed.
-                    # *(
-                    #     f"--strict-include={req.to_coord_str(versioned=False)}"
-                    #     for req in artifact_requirements
-                    #     if req.strict
-                    # ),
-                ],
-                wrapper=[bash.path, coursier.wrapper_script],
+        CoursierWrapperProcess(
+            args=(
+                coursier_report_file_name,
+                *coursier_resolve_info.coord_arg_strings,
+                # TODO(#13496): Disable --strict-include to work around Coursier issue
```

Comment:
Am I right in thinking this comment can go?

Reply:
So... assuming that we want to support versions like `3+` as mentioned in #14252, then I doubt that we'll actually be able to enable `--strict-include` anyway (unless it ignores ranges like this? but that would be odd). But the original motivation for enabling it is still valid I think... making it clearer exactly which version of a dep you were getting.

BUT, I think that now that we're able to choose the lockfile / collect all requirements without computing dependencies (via #13893), the other usecase that we had for `strict-includes` could probably use the lockfile content for inference instead: #13152.

So... yes. I think that we can forget about `--strict-include` and remove the `strict` field. But probably not in this change?